packages/clerk-js/src/utils/detect-ui-caller.ts

```diff
@@ -0,0 +1,49 @@
+function createStore(initialUsages: number) {
+  let state = initialUsages;
+
+  return {
+    get: () => state > 0,
+    increment: () => {
+      state++;
+    },
+    decrease: () => {
+      state = Math.max(state - 1, 0);
+    },
+  };
+}
+
+export const usageByUIComponents = createStore(0);
+
+const isThenable = (value: unknown): value is Promise<unknown> => {
+  return !!value && typeof (value as any).then === 'function';
+};
+
+export const makeUICaller = <T>(resource: T): T => {
```

Comment:
```suggestion
/** @internal **/ 
export const makeUICaller = <T>(resource: T): T => {
```

🙃 We could add JSDocs to indicate for external contributors that this is a tooling for internal usage